pkg/sql/util/index_util.go

```diff
@@ -284,11 +284,14 @@ func serialWithCompacted(vs []*vector.Vector, proc *process.Process) (*vector.Ve
 				}
 			}
 		case types.T_json, types.T_char, types.T_varchar, types.T_binary, types.T_varbinary, types.T_blob, types.T_text:
+			//TODO: Is T_array required here?
 			vs := vector.MustStrCol(v)
 			for i := range vs {
 				if nulls.Contains(v.GetNulls(), uint64(i)) {
 					nulls.Add(bitMap, uint64(i))
 				} else {
+
+					// TODO: If T_array is required, we might need to define arrayTypeCode in tuples.
 					ps[i].EncodeStringType([]byte(vs[i]))
```

Comment:
Hi @qingxinhome could you kindly share your feedback on this?

Reply:
Could you please check if this https://github.com/arjunsk/matrixone/pull/42 is good enough?

Reply:
I agree with your view that VECTOR and ARRAY can be limited to not being primary or unique keys. In fact, in Hive, VECTOR and ARRAY types cannot be directly used as primary keys. In PostgreSQL, VECTOR and ARRAY types can be used as part of the primary key, but cannot be directly used as the entire primary key.But you can listen to the product manager's @domingozhang opinion